### PR TITLE
Execute the update_toolbox from /tmp

### PR DESCRIPTION
**Pull request by raysiudak**
The partitions will be rearranged and reformatted, therefore
the update_toolbox should be moved to ram storage before this
happens. This change makes the executable flag and file existance
check in the init.sh script obsolete.

Contributes to EMP-273, implements EMP-387

Signed-off-by: Raymond Siudak <r.siudak@ultimaker.com>

### CODE REVIEW

File: initramfs/init.sh [hunk 1]
```diff
@@ -126,8 +135,8 @@ find_and_run_update()
 			break;
 		fi
 
-		echo "Found '${SYSTEM_UPDATE_ENTRYPOINT}' script on ${dev}, trying to execute."
-		if ! "${SYSTEM_UPDATE_ENTRYPOINT}"; then
+		echo "Found '${SYSTEM_UPDATE_ENTRYPOINT}' script on ${TOOLBOX_MOUNT}, trying to execute."
```
Comment (daid) [inline initramfs/init.sh]: Not sure if the "on TOOLBOX_MOUNT" really adds anything, just like the "on dev" didn't really tell you much (as it was already known that it was mounted from previous echo lines, and this thus tells the mount point, which doesn't really tell much)

Maybe it should just be "Found ENTRYPOINT script in toolbox, trying to execute."

Comment (raysiudak) [reply initramfs/init.sh]: Fixed, unfortunately have to approve again.

File: initramfs/init.sh [hunk 2]
```diff
@@ -106,15 +108,22 @@ find_and_run_update()
...
-			echo "Update failed: Unable to mount '${TOOLBOX_IMAGE}'."
+		echo "Found '${TOOLBOX_IMAGE}' on '${dev}', moving to tmpfs."
+		if ! mv "${UPDATE_MOUNT}/${TOOLBOX_IMAGE}" "/tmp"; then
+			echo "Update failed: Unable to move ${TOOLBOX_IMAGE} to /tmp."
+			critical_error
+			break;
+		fi
+
+		echo "Attempting to mount '/tmp/${TOOLBOX_IMAGE}' to '${TOOLBOX_MOUNT}'."
+		if ! mount "/tmp/${TOOLBOX_IMAGE}" "${TOOLBOX_MOUNT}"; then
```
[the remark this answers is not in the document]
Comment (raysiudak) [reply initramfs/init.sh]: Mount is there removal not, see #29.

File: initramfs/init.sh [hunk 3]
```diff
@@ -126,8 +135,8 @@ find_and_run_update()
 			break;
 		fi
 
-		echo "Found '${SYSTEM_UPDATE_ENTRYPOINT}' script on ${dev}, trying to execute."
-		if ! "${SYSTEM_UPDATE_ENTRYPOINT}"; then
+		echo "Found '${SYSTEM_UPDATE_ENTRYPOINT}' script, trying to execute."
+		if ! "${SYSTEM_UPDATE_ENTRYPOINT}" "${UPDATE_MOUNT}"; then
 			echo "Update failed: Error executing '${SYSTEM_UPDATE_ENTRYPOINT}'."
```
[the remark this answers is not in the document]
Comment (raysiudak) [reply initramfs/init.sh]: See #29

**Pull request merged**